codegen/src/CodeGen.scala

```diff
@@ -466,7 +466,7 @@ class CodeGen(implicit providerConfig: Config.ProviderConfig, typeMapper: TypeMa
 
     // the type has to match Pulumi's resource type schema, e.g. kubernetes:core/v1:Pod
     // please make sure, it contains 'index' instead of empty module part if needed
-    val typ = Lit.String(typeToken)
+    val typ = Lit.String(typeToken.asString)
```

Comment:
Shouldn't we use `typeToken.raw` here?

Reply:
yes, this is better, thank you

Reply:
on a second thought, if we use `raw` we cannot add the index conditionally, because the module will be empty anyway

Reply:
I think the point is we shouldn't add `index` here by ourselves in any case